### ISSUE TITLE
Improve panic reporting

### ISSUE THREAD

**Issue by flowchartsman** [magefile/mage#333]
From a UX perspective, "Error: runtime error: invalid memory address or nil pointer dereference" is not particularly helpful if I've made a mistake in my magefile that causes a panic, since there's zero context to localize it. Though I appreciate the desire to keep error message parsimonious in a build tool, it would be super helpful if I could get a full stacktrace when running with `-debug` or at least the relevant location of the panic without `-debug`

**Comment by natefinch** [comment]
yeah 100%. I think this is just not a flow I have spent a lot of time on. Just dumping the regular panic message shouldn't be too hard.

**Comment by jespino** [comment]
The server side should be already merged

**Comment by jespino** [comment]
sorry accidentally I added a comment. Already removed and removing also this one right away.

**Comment by dmotylev** [comment]
Until this lands in mage itself: I needed the same thing and built it into [magetui](https://github.com/dmotylev/magetui) — a progress-rendering library for magefiles (it wraps targets; mage itself is untouched). It recovers panics in dependency functions and renders them apart from ordinary failures: the panic value on the step's line, then the stack trimmed to start at the magefile frame, with the full untrimmed stack below it:

```text
‼ panic ▸ miscount  0.7s  panic: runtime error: index out of range [3] with length 3

  ┆ main.miscount(...)
  ┆ 	/home/me/project/magefile.go:148 +0x40

  goroutine 51 [running]:
  ... full stack, dimmed ...
```

Propagation is unchanged — the recovered panic aggregates into the error flow and exit code exactly as mage does; only the presentation differs. Not a substitute for mage printing the stack natively (you only get this for targets you wrap), but it answers "where did my magefile blow up" today.
